Lib/inspect.py

```diff
@@ -347,7 +347,10 @@ def getmembers(object, predicate=None):
         # like calling their __get__ (see bug #1785), so fall back to
         # looking in the __dict__.
         try:
-            value = getattr_static(object, key)
+            if isinstance(getattr_static(object, key), property):
```

Comment:
Why can't we always use `getattr_static`?  If there's a reason for not using it always, please describe that in a comment.

Reply:
This is related to my latest comment. Sorry, I didn't make this clear:

`getattr_static(...,'__class__')` returns `<attribute '__class__' of 'object' objects>` whereas `getattr` would return `<class '__main__.Foo'>`. What should getmembers return? Using `getattr_static` for every attribute doesn't seem to be the solution. It works fine with methods, functions and properties, but some edge cases seem to require `getattr` (such as `__class__`). It is not clear what the expected behaviour is.

Reply:
Got it, well, describe this in a comment :)